Ts refactor pt2

0xdavinchee opened the pull request.

0xdavinchee committed:
constants and helpers file extensions changed to ts

- index must be .tsx because it holds react components

0xdavinchee committed:
helpers and constants refactored

0xdavinchee committed:
used hooks file extensions changed

0xdavinchee committed:
hooks files refactored + compile

0xdavinchee committed:
store file extension changed

0xdavinchee committed:
store file refactored

- store refactored
- useAppDispatch and useAppSelector hooks added to hooks index file

0xdavinchee committed:
extra typings for existing ts slices

0xdavinchee committed:
extra typings for existing ts slices

0xdavinchee committed:
Revert "extra typings for existing ts slices"

This reverts commit c6d44196022cd33aac889399830492e9d37aeee7.

0xdavinchee committed:
slice file extensions changed

0xdavinchee committed:
slices fixed up

spaceturtleship committed:
Change MigrateThunk file extension to .ts; add typings and interfaces

spaceturtleship committed:
Change StakeThunk file extension to .ts; add typings and interfaces.

0xdavinchee committed:
do not return getBalances

0xdavinchee committed:
cleanup

0xdavinchee committed:
unused files removed

0xdavinchee committed:
Merge branch 'develop' into ts-refactor-pt0

0xdavinchee committed:
index.tsx

0xdavinchee committed:
Merge branch 'ts-refactor-pt0' into ts-refactor-pt1

0xdavinchee committed:
Merge branch 'ts-refactor-pt1' into ts-refactor-pt2

0xdavinchee committed:
jsonrpc is possible too

0xdavinchee committed:
or signer

0xdavinchee committed:
Merge branch 'ts-refactor-pt0' into ts-refactor-pt1

0xdavinchee committed:
Merge branch 'develop' into ts-refactor-pt0

0xdavinchee committed:
Merge branch 'ts-refactor-pt0' into ts-refactor-pt1

0xdavinchee committed:
Merge branch 'ts-refactor-pt1' into ts-refactor-pt2

0xdavinchee committed:
Merge branch 'develop' into ts-refactor-pt1

0xdavinchee committed:
popNum

0xdavinchee committed:
linting fix

0xdavinchee committed:
Merge branch 'ts-refactor-pt1' into ts-refactor-pt2

Zayen-X commented on src/helpers/index.tsx:
```diff
@@ -93,30 +94,37 @@ function getSohmTokenImage() {
 
-export function setAll(state, properties) {
+// TS-REFACTOR-NOTE: these may have to be any for now...
+export function setAll(state: any, properties: any) {
```
Yeah that makes sense, we _should_ have setAll for particular states. or move these out  to know what  the setAll is for.

Zayen-X commented on src/helpers/index.tsx:
```diff
@@ -93,30 +94,37 @@ function getSohmTokenImage() {
 }
 
-export const setBondState = (state, payload) => {
+export const setBondState = (state: any, payload: any) => {
```
Do we not have an interface defintion for the bond state?

0xdavinchee replied:
we actually don't use this anywhere, it's defined and used in `BondSlice.ts`.

Zayen-X commented on src/slices/MessagesSlice.ts:
```diff
@@ -35,21 +35,21 @@ const messagesSlice = createSlice({
     },
     // Finds and removes obsolete messages
-    handle_obsolete(state, action) {
+    handle_obsolete(state, _action) {
```
is `_action` unused?

0xdavinchee replied:
yup, I will just delete it

Zayen-X commented on src/slices/PendingTxnsSlice.ts:
```diff
@@ -1,23 +1,25 @@
 }
 
-const initialState: Array<any> = [];
+const initialState: Array<IPendingTxn> = [];
```
poggers.

Zayen-X commented on src/slices/StakeThunk.ts:
```diff
@@ -59,17 +67,29 @@ export const changeApproval = createAsyncThunk(
 
+interface IChangeStake {
+  action: string;
+  value: string;
```
should value be string or number?

0xdavinchee replied:
value should be a string, it's used that way both in this function and places that call `changeStake` too.

Zayen-X reviewed (changes requested):
Minor feedbacks

0xdavinchee committed:
cleanup

- no longer using fraxSlice file
- unused params just removed
- removed some unused things
- used IJsonRPCError
- await new ethers.Contract is redundant

0xdavinchee commented:
@lolchocotaco ser when pt2 merge?

Zayen-X commented:
@0xdavinchee when conflicts resolved :D sorry about that will get it done on the next conflict fix.

0xdavinchee committed:
Merge branch 'develop' into ts-refactor-pt2

0xdavinchee committed:
Merge branch 'develop' into ts-refactor-pt2
